sympy/physics/mechanics/tests/test_kane3.py

```diff
@@ -257,35 +284,54 @@ def test_bicycle():
     # many rows as *total* coordinates and speeds, but only as many columns as
     # independent coordinates and speeds.
 
-    forcing_lin = KM.linearize()[0]
-
+    A, B, _ = KM.linearize(
+        A_and_B=True,
+        op_point={
+            # Operating points for the accelerations are required for the
+            # linearizer to eliminate u' terms showing up in the coefficient
+            # matrices.
+            u1.diff(): 0,
+            u2.diff(): 0,
+            u3.diff(): 0,
+            u4.diff(): 0,
+            u5.diff(): 0,
+            u6.diff(): 0,
+            u1: 0,
+            u2: 0,
+            u3: v / PaperRadRear,
+            u4: 0,
+            u5: 0,
+            u6: v / PaperRadFront,
+            q1: 0,
+            q2: 0,
+            q4: 0,
+            q5: 0,
+        },
+        linear_solver=cramer_solve,
+    )
     # As mentioned above, the size of the linearized forcing terms is expanded
     # to include both q's and u's, so the mass matrix must have this done as
     # well.  This will likely be changed to be part of the linearized process,
     # for future reference.
-    MM_full = KM.mass_matrix_full
-
-    MM_full_s = msubs(MM_full, val_dict)
-    forcing_lin_s = msubs(forcing_lin, KM.kindiffdict(), val_dict)
+    A_s = A.xreplace(val_dict)
+    B_s = B.xreplace(val_dict)
 
-    MM_full_s = MM_full_s.evalf()
-    forcing_lin_s = forcing_lin_s.evalf()
+    A_s = A_s.evalf()
+    B_s = B_s.evalf()
```

Comment:
Is my assumption correct that `A.xreplace` and `.evalf()` are used here separately, because that is faster? It is a bit strange. If speed is required `msubs` is faster than `xreplace` (if I remember correctly).

Reply:
I'm no sure what you are pointing out here. In general, there was a deprecated use of evalf in this file and I simply swapped out use of `.evalf()` in places to get it working. I didn't try to optimize anything for speed in this file.

Reply:
Are you asking why I didn't do `A.evalf(subs=val_dict)`?

Reply:
I just tried swapping xreplace for msubs in both kane.py and linearize.py and I get NaNs in the final result. I think tried only using xreplace in kane.py, still got NaNs. Then I tried only in linearize.py which computed correctly and took the same time as with msubs (no difference in timings). msubs does a little work to try to avoid basic divide by zeros and seems to do it as fast. That bit extra msubs does seems to ensure KanesMethod gives a workable result for this problem.